tests/src/performance/perflab/CastingPerf.cs

```diff
@@ -0,0 +1,365 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Xunit.Performance;
+using System;
+using System.Collections.Generic;
+
+public interface IFoo
+{
+}
+
+public interface IFoo_1
+{
+}
+
+public interface IFoo_2
+{
+}
+
+public interface IFoo_3
+{
+}
+
+public interface IFoo_4
+{
+}
+
+public interface IFoo_5
+{
+}
+
+// C# lays the interfaces in reverse order in metadata. So IFoo is the first and IFoo_5 is last
+public class Foo : IFoo_5, IFoo_4, IFoo_3, IFoo_2, IFoo_1, IFoo
+{
+    public int m_i;
+}
+
+public class Foo_1 : Foo
+{
+    public int m_j;
+}
+
+public class Foo_2 : Foo_1
+{
+    public int m_k;
+}
+
+public class Foo_3 : Foo_2
+{
+    public int m_l;
+}
+
+public class Foo_4 : Foo_3
+{
+    public int m_m;
+}
+
+public class Foo_5 : Foo_4
+{
+    public int m_n;
+}
+
+// C# lays the interfaces in reverse order in metadata. So IFoo_1 is the first and IFoo is last
+public class Foo2 : IFoo, IFoo_5, IFoo_4, IFoo_3, IFoo_2, IFoo_1
+{
+    public int m_i;
+}
+
+public struct FooSVT
+{
+    public int m_i;
+    public int m_j;
+}
+
+public struct FooORVT
+{
+    public Object m_o;
+    public Foo m_f;
+}
+
+public interface IMyInterface1 { }
+public interface IMyInterface2 { }
+public class MyClass1 : IMyInterface1 { }
+public class MyClass2 : IMyInterface2 { }
+public class MyClass4<T> : IMyInterface1 { }
+
+public class CastingPerf
+{
+    public const int NUM_ARRAY_ELEMENTS = 100;
+
+    public static int[] j;
+    public static int[] k;
+    public static Foo[] foo;
+    public static Foo2[] foo2;
+    public static Foo[] n;
+    public static Foo_5[] foo_5;
+    public static FooSVT[] svt;
+    public static FooORVT[] orvt;
+
+    public static Object o;
+    public static Object[] o_ar;
+    public static Foo[] f;
+    public static IFoo[] ifo;
+    public static IFoo_5[] if_5;
+
+    static CastingPerf()
+    {
+        j = new int[NUM_ARRAY_ELEMENTS];
+        for (int i = 0; i < j.Length; i++)
+        {
+            j[i] = i;
+        }
+        foo = new Foo[NUM_ARRAY_ELEMENTS];
+        for (int i = 0; i < foo.Length; i++)
+        {
+            foo[i] = new Foo();
+        }
+        foo2 = new Foo2[NUM_ARRAY_ELEMENTS];
+        for (int i = 0; i < foo2.Length; i++)
+        {
+            foo2[i] = new Foo2();
+        }
+        n = new Foo[NUM_ARRAY_ELEMENTS];
+        foo_5 = new Foo_5[NUM_ARRAY_ELEMENTS];
+        for (int i = 0; i < foo_5.Length; i++)
+        {
+            foo_5[i] = new Foo_5();
+        }
+        svt = new FooSVT[NUM_ARRAY_ELEMENTS];
+        orvt = new FooORVT[NUM_ARRAY_ELEMENTS];
+    }
+
+    [Benchmark]
+    public static void ObjFooIsObj()
+    {
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                o = foo;
```

Comment:
Are we unintentionally including here costs related to writing into a static?  I'm wondering why o and foo couldn't be locals, or if the static is necessary to avoid certain JIT optimizations or something.  Same question on other such cases of this.